### DISCUSSION TITLE
Turret projectiles sometimes disappear early, cutting the turret's range by over a half

### DISCUSSION THREAD

**Discussion by NotWendy** [FakeFishGames/Barotrauma#14317] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

Every now and then, when shooting a turret, its projectiles will start to disappear almost immediately after leaving the gun. This is not a visual bug, it actually prevents enemies in the turret's line of sight from being hit.
Turret's angle doesn't matter. Once it starts happening, the gun is effectively out of commission until it decides (just as randomly) to start working again.
Happened with coilguns and lasers in my experience. Sometimes it only affects a single gun, other times multiple turrets end up being affected.
Happens in both normal levels and in the final "outpost" in the ending.
The bug is very rare. I've encountered it around 3 times over the course of 6 full campaigns. Last time it happened, it left me defenceless against the final boss and forced my group to quit that attempt.

### Reproduction steps

This one feels impossible to reproduce intentionally, with how rare it is. Sometimes you get on the gun and the projectiles just don't work. Sometimes the problem fixes itself after a bit, other time it persists for 5+ minutes, until you lose the round because you couldn't fight off enemies.

I've also encountered a similar bug in the submarine editor's test mode a while ago, where I was firing Barsuk's front laser repeatedly to test its power consumption and ended up somehow **hitting the back of my own submarine with it**. I don't know if it's even related to the other issue, but it's worth mentioning here.

### Bug prevalence

Happens every now and then

### Single player or multiplayer?

Multiplayer hosted from the in-game menu (= using a listen server)

### Version

v1.5.9.1 (Summer Update Hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by SomeRandomNoobKekeke** [comment]
On coilguns or on lasers? Because they work very differently and it's probably different bugs
Did you use funny subs or vanilla?
> 3 times over the course of 6 full campaigns

how long is that? Some stuff might have been fixed

[this](https://github.com/FakeFishGames/Barotrauma/issues/12535) have been fixed 
could be №3 from [here](https://github.com/FakeFishGames/Barotrauma/issues/13525)
[Deo's Submarine Weapon Sounds](https://new.reddit.com/r/Barotrauma/comments/1b1kpks/dont_know_why_the_coilgun_does_that_there_isnt/) mod once ate all bullets in some sector, but you said that `Turret's angle doesn't matter` so probably not the case
Could be funny performance fix effects
Also i think you can't hit jove from outside his arena coz it's too ez, mb it's bugged, mb you was lying on the floor and game considered you outside the arena

**Reply by NotWendy** [reply]
Encountered both on vanilla submarines and on a custom submarine with no modded parts. Had DE installed in one case, no mods in others. My last run happened on the latest stable version without mods and both the coilguns and the lasers were affected. Not being able to hit Jove is one thing, but I could see my projectiles disappear after leaving the gun, and guardians who were nearby didn't take damage either.
